metau-capstone/Delegates/SceneDelegate.m

```diff
@@ -22,14 +25,30 @@ - (void)scene:(UIScene *)scene willConnectToSession:(UISceneSession *)session op
     PFUser *user = [PFUser currentUser];
     if (user != nil) {
         NSLog(@"Welcome back %@ 😀", user.username);
-
-        // Load Chat view controller and set as root view controller
         UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
         UIViewController *tabBarNavigationController = [storyboard instantiateViewControllerWithIdentifier:@"tabBarController"];
         self.window.rootViewController = tabBarNavigationController;
     }
 }
 
+- (void)scene:(UIScene *)scene openURLContexts:(NSSet<UIOpenURLContext *> *)URLContexts {
+    NSURL *url = [URLContexts allObjects][0].URL;
+    NSArray *queryPair = [url.query componentsSeparatedByString:@"="];
+    NSString *userID = queryPair[1];
+    // Construct Query for Flashcards
+    PFQuery *query = [PFQuery queryWithClassName:@"Flashcard"];
+    [query whereKey:@"userID" equalTo:userID];
+    [query findObjectsInBackgroundWithBlock:^(NSArray<Flashcard *> *cards, NSError * _Nullable error) {
+        if (!error) {
+            // Create array of Preview Cards
+            [APIManager shared].previewFlashcards = [PreviewFlashcard createCardsFromArray:cards];
+            // Set view controller to preview view controller
+            UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+            UINavigationController *previewNavigationController = [storyboard instantiateViewControllerWithIdentifier:@"PreviewNavigationController"];
+            self.window.rootViewController = previewNavigationController;
+        }
```

Comment:
Good polish will be handling the error case!

What are you think about when an error is returned?